### DISCUSSION TITLE
Failed to create session

### DISCUSSION THREAD

**Discussion by ashinkraju** [appium/appium#17149] [Q&A]
An unknown server-side error occurred while processing the command. Original error: Cannot verify the signature of 'C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk'. Original error: The 'java.exe' binary could not be found neither in PATH nor under JAVA_HOME (C:\Program Files\Java\jdk-12\bin\bin)

How to solve this error?

Appium Server Log
----------------------
[info] [35m[HTTP][39m Waiting until the server is closed
[info] [35m[HTTP][39m Received server close event
[info] [35m[Appium][39m Welcome to Appium v1.22.3
[info] [35m[Appium][39m Non-default server args:
[info] [35m[Appium][39m   relaxedSecurityEnabled: true
[info] [35m[Appium][39m   allowInsecure: {
[info] [35m[Appium][39m   }
[info] [35m[Appium][39m   denyInsecure: {
[info] [35m[Appium][39m   }
[info] [35m[Appium][39m Appium REST http interface listener started on 0.0.0.0:4723[info] [35m[HTTP][39m [37m-->[39m [37mPOST[39m [37m/wd/hub/session[39m
[info] [35m[HTTP][39m [90m{"capabilities":{"alwaysMatch":{"appium:AUTOMATION_NAME":"Appium","platformName":"Android","appium:PLATFORM_VERSION":"12.0","appium:DEVICE_NAME":"Android","appium:appPackage":"com.google.android.calculator","appium:appActivity":"com.android.calculator2.Calculator","appium:ensureWebviewsHavePages":true,"appium:nativeWebScreenshot":true,"appium:newCommandTimeout":3600,"appium:connectHardwareKeyboard":true},"firstMatch":[{}]},"desiredCapabilities":{"appium:AUTOMATION_NAME":"Appium","platformName":"Android","appium:PLATFORM_VERSION":"12.0","appium:DEVICE_NAME":"Android","appium:appPackage":"com.google.android.calculator","appium:appActivity":"com.android.calculator2.Calculator","appium:ensureWebviewsHavePages":true,"appium:nativeWebScreenshot":true,"appium:newCommandTimeout":3600,"appium:connectHardwareKeyboard":true}}[39m
[debug] [35m[W3C][39m Calling AppiumDriver.createSession() with args: [{"appium:AUTOMATION_NAME":"Appium","platformName":"Android","appium:PLATFORM_VERSION":"12.0","appium:DEVICE_NAME":"Android","appium:appPackage":"com.google.android.calculator","appium:appActivity":"com.android.calculator2.Calculator","appium:ensureWebviewsHavePages":true,"appium:nativeWebScreenshot":true,"appium:newCommandTimeout":3600,"appium:connectHardwareKeyboard":true},null,{"alwaysMatch":{"appium:AUTOMATION_NAME":"Appium","platformName":"Android","appium:PLATFORM_VERSION":"12.0","appium:DEVICE_NAME":"Android","appium:appPackage":"com.google.android.calculator","appium:appActivity":"com.android.calculator2.Calculator","appium:ensureWebviewsHavePages":true,"appium:nativeWebScreenshot":true,"appium:newCommandTimeout":3600,"appium:connectHardwareKeyboard":true},"firstMatch":[{}]}]
[debug] [35m[BaseDriver][39m Event 'newSessionRequested' logged at 1657251429405 (09:07:09 GMT+0530 (India Standard Time))
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m ======================================================================
[warn] [35m[Appium][39m   DEPRECATION WARNING:
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m   The 'automationName' capability was not provided in the desired 
[warn] [35m[Appium][39m   capabilities for this Android session
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m   Setting 'automationName=UiAutomator2' by default and using the 
[warn] [35m[Appium][39m   UiAutomator2 Driver
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m   The next major version of Appium (2.x) will **require** the 
[warn] [35m[Appium][39m   'automationName' capability to be set for all sessions on all 
[warn] [35m[Appium][39m   platforms
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m   In previous versions (Appium <= 1.13.x), the default was 
[warn] [35m[Appium][39m   'automationName=UiAutomator1'
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m   If you wish to use that automation instead of UiAutomator2, please 
[warn] [35m[Appium][39m   add 'automationName=UiAutomator1' to your desired capabilities
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m   For more information about drivers, please visit 
[warn] [35m[Appium][39m   http://appium.io/docs/en/about-appium/intro/ and explore the 
[warn] [35m[Appium][39m   'Drivers' menu
[warn] [35m[Appium][39m 
[warn] [35m[Appium][39m ======================================================================
[warn] [35m[Appium][39m 
[info] [35m[Appium][39m Appium v1.22.3 creating new AndroidUiautomator2Driver (v1.70.1) session
[info] [35m[Appium][39m Applying relaxed security to 'AndroidUiautomator2Driver' as per server command line argument. All insecure features will be enabled unless explicitly disabled by --deny-insecure
[debug] [35m[BaseDriver][39m W3C capabilities and MJSONWP desired capabilities were provided
[debug] [35m[BaseDriver][39m Creating session with W3C capabilities: {
[debug] [35m[BaseDriver][39m   "alwaysMatch": {
[debug] [35m[BaseDriver][39m     "platformName": "Android",
[debug] [35m[BaseDriver][39m     "appium:AUTOMATION_NAME": "Appium",
[debug] [35m[BaseDriver][39m     "appium:PLATFORM_VERSION": "12.0",
[debug] [35m[BaseDriver][39m     "appium:DEVICE_NAME": "Android",
[debug] [35m[BaseDriver][39m     "appium:appPackage": "com.google.android.calculator",
[debug] [35m[BaseDriver][39m     "appium:appActivity": "com.android.calculator2.Calculator",
[debug] [35m[BaseDriver][39m     "appium:ensureWebviewsHavePages": true,
[debug] [35m[BaseDriver][39m     "appium:nativeWebScreenshot": true,
[debug] [35m[BaseDriver][39m     "appium:newCommandTimeout": 3600,
[debug] [35m[BaseDriver][39m     "appium:connectHardwareKeyboard": true
[debug] [35m[BaseDriver][39m   },
[debug] [35m[BaseDriver][39m   "firstMatch": [
[debug] [35m[BaseDriver][39m     {}
[debug] [35m[BaseDriver][39m   ]
[debug] [35m[BaseDriver][39m }
[warn] [35m[BaseDriver][39m The following capabilities were provided, but are not recognized by Appium:
[warn] [35m[BaseDriver][39m   AUTOMATION_NAME
[warn] [35m[BaseDriver][39m   PLATFORM_VERSION
[warn] [35m[BaseDriver][39m   DEVICE_NAME
[warn] [35m[BaseDriver][39m   connectHardwareKeyboard
[info] [35m[BaseDriver][39m Session created with session id: f9c8ff84-cacc-4f24-b1b8-2da918c07d61
[info] [35m[UiAutomator2][39m Starting 'com.google.android.calculator' directly on the device[info] [35m[AndroidDriver][39m Retrieving device list
[debug] [35m[ADB][39m Trying to find a connected android device
[debug] [35m[ADB][39m Getting connected devices[debug] [35m[ADB][39m Connected devices: [{"udid":"emulator-5554","state":"device"}]
[info] [35m[AndroidDriver][39m Using device: emulator-5554
[info] [35m[ADB][39m Using 'adb.exe' from 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe'
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 start-server'[debug] [35m[ADB][39m Setting device id to emulator-5554
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell getprop ro.build.version.sdk'[debug] [35m[ADB][39m Current device property 'ro.build.version.sdk': 31
[info] [35m[ADB][39m Getting device platform version
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell getprop ro.build.version.release'[debug] [35m[ADB][39m Current device property 'ro.build.version.release': 12
[debug] [35m[ADB][39m Device API level: 31
[info] [35m[UiAutomator2][39m Relaxing hidden api policy
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell 'settings put global hidden_api_policy_pre_p_apps 1;settings put global hidden_api_policy_p_apps 1;settings put global hidden_api_policy 1''[warn] [35m[AndroidDriver][39m No app sent in, not parsing package/activity
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 wait-for-device'[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell echo ping'[debug] [35m[AndroidDriver][39m Pushing settings apk to device...
[debug] [35m[ADB][39m Getting install status for io.appium.settings
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell dumpsys package io.appium.settings'[debug] [35m[ADB][39m 'io.appium.settings' is installed
[debug] [35m[ADB][39m Getting package info for 'io.appium.settings'
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell dumpsys package io.appium.settings'[debug] [35m[ADB][39m The version name of the installed 'io.appium.settings' is greater or equal to the application version name ('3.4.0' >= '3.4.0')
[debug] [35m[ADB][39m There is no need to install/upgrade 'C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\io.appium.settings\apks\settings_apk-debug.apk'
[debug] [35m[ADB][39m Getting IDs of all 'io.appium.settings' processes
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell 'pgrep --help; echo $?''[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'[debug] [35m[AndroidDriver][39m io.appium.settings is already running. There is no need to reset its permissions.
[debug] [35m[Logcat][39m Starting logs capture with command: C:\\Users\\THIC2111768.THIME\\AppData\\Local\\Android\\Sdk\\platform-tools\\adb.exe -P 5037 -s emulator-5554 logcat -v threadtime[debug] [35m[UiAutomator2][39m Forwarding UiAutomator2 Server port 6790 to local port 8200[debug] [35m[ADB][39m Forwarding system: 8200 to device: 6790
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 forward tcp:8200 tcp:6790'[debug] [35m[ADB][39m Getting install status for io.appium.uiautomator2.server
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell dumpsys package io.appium.uiautomator2.server'[debug] [35m[ADB][39m 'io.appium.uiautomator2.server' is not installed
[debug] [35m[ADB][39m App 'C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk' is not installed
[debug] [35m[UiAutomator2][39m io.appium.uiautomator2.server installation state: notInstalled
[debug] [35m[ADB][39m Checking app cert for C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk[info] [35m[ADB][39m Using 'apksigner.jar' from 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\build-tools\33.0.0\lib\apksigner.jar'
[debug] [35m[UiAutomator2][39m Deleting UiAutomator2 session
[debug] [35m[UiAutomator2][39m Deleting UiAutomator2 server session
[debug] [35m[WD Proxy][39m Matched '/' to command name 'deleteSession'
[warn] [35m[UiAutomator2][39m Did not get confirmation UiAutomator2 deleteSession worked; Error was: UnknownError: An unknown server-side error occurred while processing the command. Original error: Trying to proxy a session command without session id
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell am force-stop com.google.android.calculator'[debug] [35m[Logcat][39m Stopping logcat capture
[debug] [35m[ADB][39m Removing forwarded port socket connection: 8200 
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 forward --remove tcp:8200'[info] [35m[UiAutomator2][39m Restoring hidden api policy to the device default configuration
[debug] [35m[ADB][39m Running 'C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell 'settings delete global hidden_api_policy_pre_p_apps;settings delete global hidden_api_policy_p_apps;settings delete global hidden_api_policy''[debug] [35m[BaseDriver][39m Event 'newSessionStarted' logged at 1657251438511 (09:07:18 GMT+0530 (India Standard Time))
[debug] [35m[W3C][39m Encountered internal error running command: Error: Cannot verify the signature of 'C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-server\apks\appium-uiautomator2-server-v4.27.0.apk'. Original error: The 'java.exe' binary could not be found neither in PATH nor under JAVA_HOME (C:\Program Files\Java\jdk-12\bin\bin)
[debug] [35m[W3C][39m     at ADB.checkApkCert (C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-adb\lib\tools\apk-signing.js:306:11)
[debug] [35m[W3C][39m     at UiAutomator2Server.installServerApk (C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\uiautomator2.js:117:13)
[debug] [35m[W3C][39m     at AndroidUiautomator2Driver.initUiAutomator2Server (C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\driver.js:494:7)
[debug] [35m[W3C][39m     at AndroidUiautomator2Driver.startUiAutomator2Session (C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\driver.js:390:5)
[debug] [35m[W3C][39m     at AndroidUiautomator2Driver.createSession (C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-uiautomator2-driver\lib\driver.js:229:7)
[debug] [35m[W3C][39m     at AppiumDriver.createSession (C:\Users\THIC2111768.THIME\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\lib\appium.js:387:35)
[info] [35m[HTTP][39m [37m<-- POST /wd/hub/session [39m[31m500[39m [90m9124 ms - 1401[39m
[info] [35m[HTTP][39m [90m[39m

Installed: Java 12
                Appium Desktop 1.22.3-4
                Appium Inspector 2022.5.4

**Comment by KazuCocoa** [comment]
What the result of PATH or JAVA_HOME?

**Reply by ashinkraju** [reply]
C:\Program Files\Java\jdk-12

**Reply by ashinkraju** [reply]
ANDROID_HOME :  C:\Users\THIC2111768.THIME\AppData\Local\Android\Sdk
JAVA_HOME : C:\Program Files\Java\jdk-12\bin

**Reply by jlipps** [reply]
JAVA_HOME should not include the bin dir. You can clearly see how this is causing an error in the log (note the doubled `bin`):

```
Original error: The 'java.exe' binary could not be found neither in PATH nor under JAVA_HOME (C:\Program Files\Java\jdk-12\bin\bin
```

**Reply by ashinkraju** [reply]
Ok let me try

**Reply by ashinkraju** [reply]
Solved Thank u
